python/lsst/obs/base/cli/butler_cmd_test.py

```diff
@@ -36,24 +36,41 @@ class ButlerCmdTestBase(metaclass=abc.ABCMeta):
     Subclass from this, then `unittest.TestCase` to get a working test suite.
     """
 
-    instrument_class = None
-    """The fully qualified instrument class.
-    """
+    @staticmethod
+    @abc.abstractmethod
+    def instrumentClass():
```

Comment:
These do indeed force a test implementation to specify the value but in most cases the tests will fail if they do it wrong. It's a shame that a subclass can't implement by doing a simple class property but the use of static method here prevents that.

Also instrumentClass as a name makes you think this is the class but it's not, it's the class name.